ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -437,16 +437,22 @@ ledgerDbPush cfg ap db =
       applyBlock (ledgerDbCfg cfg) ap db
 
 -- | Push a bunch of blocks (oldest first)
-ledgerDbPushMany :: (ApplyBlock l blk, Monad m, c)
-                 => (UpdateLedgerDbTraceEvent blk -> m ())
-                 -> LedgerDbCfg l
-                 -> [Ap m l blk c] -> LedgerDB l -> m (LedgerDB l)
-ledgerDbPushMany trace = repeatedlyM . pushAndTrace
+ledgerDbPushMany ::
+     forall m c l blk . (ApplyBlock l blk, Monad m, c)
+  => (UpdateLedgerDbTraceEvent blk -> m ())
+  -> LedgerDbCfg l
+  -> [Ap m l blk c] -> LedgerDB l -> m (LedgerDB l)
+ledgerDbPushMany trace cfg aps initDb = (repeatedlyM pushAndTrace) aps initDb
   where
-    pushAndTrace cfg ap db = do
-      trace $ StartedPushingBlockToTheLedgerDb $ toRealPoint ap
-      res <- ledgerDbPush cfg ap db
-      return res
+    pushAndTrace ap db = do
+      traceStep $ StartedPushingBlockToTheLedgerDb (Pushing $ toRealPoint ap)
```

Comment:
What about something more direct (with the use of auxiliary `mk` functions)? 

```haskell
trace $ tartedPushingBlockToTheLedgerDb $ (mkPushing ap ) (mkPushGoal aps)
```

Reply:
I like it